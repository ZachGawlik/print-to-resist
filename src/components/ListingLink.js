import PropTypes from 'prop-types';
import React from 'react';
import { Link } from 'react-router-dom';

const ListingLink = ({ children, listingId }) =>
  <Link className="ListingLink" to={`/poster/${listingId}`}>
    {children}
  </Link>;

ListingLink.propTypes = {
  children: PropTypes.node.isRequired,
  listingId: PropTypes.number.isRequired
};

export default ListingLink;
