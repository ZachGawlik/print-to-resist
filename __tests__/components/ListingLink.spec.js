import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import ListingLink from '../../src/components/ListingLink';

describe('ListingLink', () => {
  it('renders as expected', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<ListingLink listingId={5}>Go to listing</ListingLink>);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
