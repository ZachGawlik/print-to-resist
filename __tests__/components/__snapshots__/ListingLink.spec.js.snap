// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListingLink renders as expected 1`] = `
<Link
  className="ListingLink"
  replace={false}
  to="/poster/5"
>
  Go to listing
</Link>
`;
